### PR TITLE
feature/ICL-96 configure cache server version

### PR DESCRIPTION
**Pull request by JGiter**
Should be merged after https://github.com/energywebfoundation/iam-cache-server/pull/162

### CODE REVIEW

File: src/cacheServerClient/cacheServerClient.ts [hunk 1]
```diff
@@ -33,9 +36,13 @@ export class CacheServerClient implements ICacheServerClient {
...
 
-    constructor({ url, cacheServerSupportsAuth = true }: CacheServerClientOptions, signer: Signer) {
+    constructor(
+        { url, cacheServerSupportsAuth = true, apiVersion = defaultCacheServerApiVersion }: CacheServerClientOptions,
+        signer: Signer,
+    ) {
+        console.log(`CacheServerClient.constructor url: ${url}, apiVersion: ${apiVersion}`);
         this.httpClient = axios.create({
-            baseURL: url,
+            baseURL: url.endsWith("/") ? `${url}${apiVersion}` : `${url}/${apiVersion}`,
```
Comment (jrhender) [inline src/cacheServerClient/cacheServerClient.ts:45]: I'm just curious what you see as the advantage of having a separate `apiVersion` param over having folks add the version to the `url`?

**Review by jrhender — CHANGES_REQUESTED** [review]
I think the commit should not start with `refactor` as I think this will not trigger a package release (unless we've configured it to do so): https://github.com/semantic-release/commit-analyzer

File: src/cacheServerClient/cacheServerClient.ts (hunk 1 above)
Comment (JGiter) [reply src/cacheServerClient/cacheServerClient.ts]: Separation of version from base url would give more flexibility like changing type of versioning, but it is very unlikely be needed. So probably keeping them together is simpler

Comment (jrhender) [reply src/cacheServerClient/cacheServerClient.ts]: Ok, so is this PR not needed then as we can simply change the `url`?

**Comment by JGiter** [comment]
> I think the commit should not start with `refactor` as I think this will not trigger a package release (unless we've configured it to do so): https://github.com/semantic-release/commit-analyzer

Yes, we configured it already https://github.com/energywebfoundation/iam-client-lib/blob/9a55b466aff5803c6e9e6e9cb2a041e32872a8e3/.releaserc.json#L21

File: src/cacheServerClient/cacheServerClient.ts (hunk 1 above)
Comment (jrhender) [reply src/cacheServerClient/cacheServerClient.ts]: Whoops, nevermind, I see you've changed this

**Review by jrhender — APPROVED** [review]
Looks good!

**Pull request merged**